src/deepsparse/pipeline_loggers/manager.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A manager that oversees all the pipeline loggers
+"""
+
+from typing import Any, Dict, List, Set, Union
+
+from deepsparse import PipelineLogger
+from deepsparse.pipeline_loggers.prometheus_pipeline_logger import (
+    IDENTIFIER as PROMETHEUS_IDENTIFIER,
+)
+from deepsparse.pipeline_loggers.prometheus_pipeline_logger import PrometheusLogger
+from deepsparse.timing.timing_schema import InferenceTimingSchema
+
+
+__all__ = ["LoggerManager"]
+
+SUPPORTED_LOGGERS = {PROMETHEUS_IDENTIFIER}
+
+
+class LoggerManager:
+    """
+    Object that contains multiple loggers for
+    the given inference pipeline.
+
+    The envisioned lifecycle of a logger
+    manager:
+
+    ```
+    pipeline = ... # define a pipeline
+    pipeline_name = pipeline.name # fetch the pipeline name
+
+    # create a LoggerManager
+    logger_manager = LoggerManager("logger_name")
+    or
+    logger_manager = LoggerManager(["logger_name_1", "logger_name_2", ...])
+
+    # log the data for the particular inference pipeline
+
+    timings, data = ...
+    logger.log_timings(pipeline_name, timings)
+    logger.log_data(pipeline_name, data)
+
+    ```
+
+    :param logger_identifiers: The identifier(s) of the logger types that
+        will be created within the scope of the manager. This can be either a
+        single identifier, or a list of identifiers (for multiple loggers).
+    :param supported_loggers: A set of supported logger identifiers; listed in
+        `SUPPORTED_LOGGERS` variable
+    """
+
+    def __init__(
+        self,
+        logger_identifiers: Union[str, List[str]],
+        supported_loggers: Set[str] = SUPPORTED_LOGGERS,
+    ):
+        self._supported_loggers = supported_loggers
```

Comment:
Are we expecting people to pass in their own set of supported loggers? I think this should not be a parameter and should just be a instance variable.

Additionally if you make the identifier a abstract class property you could do:
```python
loggers = [PrometheusLogger]
self._supported_loggers: Dict[str, Type[PipelineLogger]] = {logger.identifier: logger}
```

Reply:
In general, your whole set of bits of advice helps tremendously. However, the code above would not work. Properties are "properties" (duh!) of instances and not classes. There is a hack to make them global though, will try that since I really like the simplifications you are introducing